src/components/buttons/Button.vue

```diff
@@ -1,5 +1,18 @@
 <template>
+  <a
+    v-if="href"
+    :href="href"
+    :class="[
+      'button',
+      { 'button--primary': primary, [`button--${size}`]: size },
+    ]"
+    :style="style"
+    @click="onClick"
+  >
+    <slot />
+  </a>
   <button
+    v-else
```

Comment:
@antoninroam @jpyepez-roam I tried setting this up as a dynamic Vue component with `<component :is="someFunc">` where the function would return the tagname, so either `a` or `button` depending on if `href` was defined or not - but it didn't work. 

Do any of you know if it's possible to render dynamic tags with Vue this way? I just opted for this current solution instead for now. 

https://v3.vuejs.org/guide/component-dynamic-async.html#dynamic-async-components

Reply:
maybe you could use named slots for that? Looks similar to what you have currently, but a bit more readable, I guess.
https://v3.vuejs.org/guide/component-slots.html#named-slots
Sorry, I'm not much of a help here, didn't have anything like that in my project 😅

Reply:
The docs kinda present this type of issue (but way more exaggerated). Either render's `h` function (createElement) or JSX tend to address this as one of the template approaches' shortcomings.

https://v3.vuejs.org/guide/render-function.html#the-dom-tree

Hope it helps!